framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -975,11 +990,13 @@ public void checkMethod(MethodTree methodDeclTree, AnnotatedExecutableType type)
      * @param methodType annotated type of the method
      */
     private void checkTypeVariables(MethodTree node, AnnotatedExecutableType methodType) {
-        JavaExpressionContext context =
-                JavaExpressionContext.buildContextForMethodDeclaration(node, factory.getChecker());
         for (int i = 0; i < methodType.getTypeVariables().size(); i++) {
             AnnotatedTypeMirror atm = methodType.getTypeVariables().get(i);
-            viewpointAdaptToContext(context, atm);
+            convertAnnotatedTypeMirror(
+                    stringExpr ->
+                            StringToJavaExpression.atMethodBody(
+                                    stringExpr, node, factory.getChecker()),
+                    atm);
             checkType(atm, node.getTypeParameters().get(i));
         }
     }
```

Comment:
In `ExpressionErrorChecker`, abstract out the list concatenation into a named method (or find one that already exists).  Also, for efficiency it should check each argument (`r1` and `r2`) with `.isEmpty()` and if so, return the other one without allocating a new list.  When it does need to concatenate, it should size the list appropriately.